jzaia_files/main.cpp

```diff
@@ -0,0 +1,225 @@
+#include <algorithm>
+#include <cassert>
+#include <chrono>
+#include <cstddef>
+#include <iostream>
+#include <vector>
+
+#include "Gates.hpp"
+#include "GateImplementationsLM.hpp"
+#include "MeasurementsLQubit.hpp"
+#include "ObservablesLQubit.hpp"
+#include "StateVectorLQubitManaged.hpp"
+
+
+using Pennylane::LightningQubit::StateVectorLQubitManaged;
+using namespace Pennylane::Gates;
+using namespace Pennylane::LightningQubit;
+using namespace Pennylane::LightningQubit::Measures;
+using namespace Pennylane::LightningQubit::Observables;
+
+// Define values to be selected by each oracle
+#define ORACLE1_QUBITS (6)
+/* Oracle 1 selects the string: "11010" */
+#define ORACLE1_EXPECTED (std::vector<bool>{true, true, false, true, false})
+
+#define ORACLE2_QUBITS (10)
+/* Oracle 2 selects the string: "101010101" */
+#define ORACLE2_EXPECTED (std::vector<bool>{true, false, true, false, true, \
+                                            false, true, false, true})
+
+#define ORACLE3_QUBITS (17)
+/* Oracle 2 selects the string: "0011001100110011" */
+#define ORACLE3_EXPECTED (std::vector<bool>{false, false, true, true, false, \
+                                            false, true, true, false, false, \
+                                            true, true, false, false, true, \
+                                            true})
```

Comment:
Could you do this without macros?

Reply:
Absolutely, I changed these to global consts in ff86a5e, is that what you meant? Or do you prefer to stay away from those as well and just define these directly in main? I know sometimes formatting guidelines are opinionated on whether or not globals are good or bad design, but I'm pretty neutral on them.